cogs/displayinfo.py

```diff
@@ -33,85 +32,139 @@ class DisplayInfo(commands.Cog):
-                            try:
-                                await self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"]
-                            except asyncio.CancelledError:
-                                print("task is now cancelled")
-                        self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"] = None
-
-                        currentStartTime = self.timers[mem_before.name][self.current_date][mem_before.activity.name]["startTime"]
-                        if currentStartTime:
-                            playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
-                            self.timers[mem_before.name][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[mem_before.name][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
-                        print("end:", self.timers) # printing the dictionary to see the updates after an activity is ended
-                        self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
-
-    # This function is an asynchronous coroutine that runs indefinitely in the background when mem_after.activity is True.
-    # It is designed to update activity time information for a Discord member ('mem_after') whenever they start a new activity every 5 minutes.
-    async def timed_update(self, mem_before: discord.Member, mem_after: discord.Member):
-        await asyncio.sleep(300)
-        while not mem_before.activity and mem_after.activity:
-            playTime = 300 # add 5 mins to activity time
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["totalTime"] = self.timers[mem_after.name][self.current_date][mem_after.activity.name].get("totalTime", 0) + playTime
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now()
-            print("in timed_update", self.timers) #printing the dictionary to see the timed updates
+        if mem_after.status == discord.Status.online and (not mem_before.activity and mem_after.activity): # activity starts/ongoing
+            if mem_after.id not in self.members_online:
+                self.members_online[mem_after.id] = mem_after.guild.id
+                await self.reset_start(mem_after)
+
+                if self.timed_task == None: #if task is finished create new task
+                    self.timed_task = asyncio.create_task(self.timed_update())
+                print("added:", mem_after.name)
+            
+            if self.members_online[mem_after.id] == mem_after.guild.id:
```

Comment:
Might be better to swap this conditional to `mem_after.guild.id == self.members_online[mem_after.id]` for readability so that we know we are trying to find the right guild to match with what we previously set in `self.members_online`